File: telerik_ui_for_winforms_courseware_chm/Courseware/RealEstate/CS/DockUserControls/CalendarPanel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace DockUserControls
{
    public partial class CalendarPanel : UserControl
    {
        public CalendarPanel()
        {
            InitializeComponent();
        }

        public delegate void DateChangedHandler(DateTime date);
        public event DateChangedHandler DateChanged;

        private void radCalendar1_SelectionChanged(object sender, EventArgs e)
        {
            if (DateChanged != null)
            {
                DateChanged((sender as Telerik.WinControls.UI.RadCalendar).SelectedDate);
            }
        }
    }
}

File: telerik_ui_for_winforms_courseware_chm/Courseware/RealEstate/VB/DockUserControls/CalendarPanel.vb
Imports Telerik.WinControls.UI

Public Class CalendarPanel

    Public Delegate Sub DateChangedHandler(ByVal [date] As DateTime)
    Public Event DateChanged As DateChangedHandler

    Private Sub RadCalendar1_SelectionChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles RadCalendar1.SelectionChanged
        RaiseEvent DateChanged((TryCast(sender, RadCalendar)).SelectedDate)
    End Sub
End Class
